src/fielddef.ts

```diff
@@ -265,6 +265,11 @@ export function normalize(fieldDef: ChannelDef, channel: Channel) {
     if (!compatible) {
       log.warn(warning);
     }
+    if (fieldDef.type === 'ordinal') {
+      if ((<PositionFieldDef>fieldDef).axis.format) {
```

Comment:
Yeah in general avoid casting because its usually wrong.

This seems like the wrong location to add this warning.  Maybe initAxis in unit.ts and facet.ts.

You should also add test that consumes the warning using log.wrap